python/ray/data/tests/test_consumption.py

```diff
@@ -158,6 +158,19 @@ def test_count_edge_case(ray_start_regular):
     assert actual_count == 5
 
 
+def test_count_after_partial_execution(ray_start_regular):
+    paths = ["example://iris.csv"] * 5
+    ds = ray.data.read_csv(paths, override_num_blocks=15)
```

Comment:
Nit: What happens if we don't specify `override_num_blocks` here? If it's unnecessary, could be good to remove it. Writing minimally passing tests is a good practice